nox/_parametrize.py

```diff
@@ -13,9 +13,59 @@
 # limitations under the License.
 
 import functools
+import itertools
 
 
-def parametrize_decorator(arg_names, arg_values_list):
+class Param:
+    """A class that encapsulates a single set of parameters to a parametrized
+    session."""
+
+    def __init__(self, *args, arg_names=None, id=None):
+        self.args = tuple(args)
+        self.id = id
```

Comment:
I am 👎 on locals (here `id`) shadowing `__builtins__`. You could get around this by using `kwargs.get('id')` if you really want `id` as the argument name or just using `id_`. Using `id` as an attribute is less an issue but I also try to avoid if I can.

Reply:
I'm trying to match pytest here and since this is user-surfaced, I don't want to do `id_`.